### PR TITLE
Add linter and fix lint issues

### PR DESCRIPTION
**Pull request by RMPall**
### What

This pr attempts to enable golangcilint in this repo and fix all the existing lint issues

### How to review

Please check if the changes make sense and all the expected linters are enabled. 

### Who can review

Anyone except me

### CODE REVIEW

File: config_security.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"crypto/x509"
 	"errors"
 	"fmt"
-	"io/ioutil"
+	"os"
```
Comment (nshumoogum) [inline config_security.go:8]: Do we not use io instead of io/ioutil? Just for consistency instead of using os?

File: global.go [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package kafka
 
 import (
+	"crypto/rand"
 	"math"
-	"math/rand"
```
Comment (nshumoogum) [inline global.go:5]: Why switching out `math/rand` with `crypto/rand` ?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: global.go (hunk 2 above)
Comment (rahulPalliyalil) [reply global.go:5]: use of crypto rand is suggested by the gosec linter as math rand is considered weaker random number generator.

File: config_security.go (hunk 1 above)
Comment (rahulPalliyalil) [reply config_security.go:8]: ioutil package is depreciated  after 1.18, so that's why used os instead of ioutil.

**Review by nshumoogum — DISMISSED** [review]

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**